### DISCUSSION TITLE
Rescan does not recover faces removed via Edit Dialogue

### DISCUSSION THREAD

**Discussion by rpanizzon** [photoprism/photoprism#2817] [Q&A]
#### 1. Rescan of Library does not find faces that have been removed via the Edit Dialogue

I have accidentally deleted several faces via the edit dialogue. 
I am now trying to get them back. I have tried rescanning the photos (with and without "Complete Rescan") to recover the faces. However, the rescan does not find "new faces".

#### 2. How can we reproduce it? 
On a photo with 1 or more faces:

1. Go to Edit Dialogue (Edit symbol on top right). 
2. Click on "PEOPLE"
3. If the face is "recognised", clear it by pressing on the up arrow.
4. Press the "x" in the top right corner of the face. 
5. Image no longer recognise the face.
6. Go to "Library" and do a "Complete Rescan" of the  folder containing the image
7. After some time for face recognition to reprocess,  go back to the image in search.  
8. Click on image and go to "Edit Dialogue"
9. The face is still missing - not recovered

#### 3. What behavior do you expect?
I expected that a "Complete rescan" will find the face again.
 
#### 4. What could be the cause of your problem?
Rescan does not reset if the Edit Dialogue has deleted the face.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?
If required.

#### 6. Which software versions do you use?
PhotoPrism® 220121-2b4c8e1f-Linux-x86_64

**Comment by graciousgrey** [comment]
Indexing and complete indexing won't undo manual changes. This is the expected behaviour as people don't want their manual edits to be gone when re-indexing.

Undoing a removal of faces is only possible directly after removing them as described here: https://docs.photoprism.app/user-guide/organize/people/#remove-faces

**Reply by rpanizzon** [reply]
So effectively, there is no way of undoing a removal of a face after a reload?
This is unfortunate as it means I'll have to start again from scratch....

Suggest the documentation may need a little bit more clarity around both edit dialogue and deleting a true face (non recoverable), and add it to the reasons a face may not be detected.
I would not have deleted any faces if I thought it was irrecoverable

**Reply by lastzero** [reply]
You could update the database table manually or wait until we offer manual tagging. Best way for that is to become a sponsor.

**Reply by lastzero** [reply]
We suggest you contribute to the documentaton!

**Reply by graciousgrey** [reply]
As described in our docs the remove face functionality has been built to remove false positives or faces one don't want to keep. If you just want to hide faces temporarily use the hide functionality: https://docs.photoprism.app/user-guide/organize/people/#hiding-people

**Reply by rpanizzon** [reply]
Thanks lastzero.
I'm fairly new to Photoprism, so not quite at the sponsor level yet. Happy to make updates to the documentation once I get a little bit more familiar with the product.
Appreciate your quick reply and the great work you guys are doing.
